axelrod/tests/strategies/test_axelrod_second.py

```diff
@@ -694,3 +695,162 @@ def test_strategy(self):
+        actions += [(C, D)]
+        # Opponent's defect increments even streak to 5, so we cooperate.
+        actions += [(C, C)]
+        actions += [(C, D), (D, C), (C, D), (D, C), (C, D)]
+        # Another 5 streak
+        actions += [(C, C)]
+        # Repeat
+        actions += [(C, D), (D, C), (C, D), (D, C), (C, D), (C, C)] * 3
+        # Repeat.  Notice that the last turn is the 37th move, but we do not
+        # defect.
+        actions += [(C, D), (D, C), (C, D), (D, C), (C, D), (C, C)]
+        self.versus_test(axelrod.Alternator(), expected_actions=actions)
+
+        # Test for parity limit shortening.
+        opponent_actions = [D, C] * 1000
+        AsyncAlternator = axelrod.MockPlayer(actions=opponent_actions)
+        actions = [(C, D), (D, C), (C, D), (D, C), (C, D), (C, C)] * 6
+        # Defect on 37th move
+        actions += [(D, D)]
+        actions += [(C, C)]
+        # This triggers the burned flag.  We should just Tit-for-Tat from here.
+        actions += [(C, D)]
+        actions += [(D, C), (C, D), (D, C), (C, D), (C, C)]
+        # This is the seventh time we've hit the limit.  So do it once more.
+        actions += [(C, D), (D, C), (C, D), (D, C), (C, D), (C, C)]
+        # Now hit thi limit sooner
+        actions += [(C, D), (D, C), (C, D), (C, C)] * 5
+        self.versus_test(AsyncAlternator, expected_actions=actions, attrs={"parity_limit": 3})
+
+        # Use a Defector to test the 20-defect streak
+        actions = [(C, D), (D, D), (D, D), (D, D), (D, D)]
+        # Now the two parity flags are used
+        actions += [(C, D), (C, D)]
+        # Repeat
+        actions += [(D, D), (D, D), (D, D), (D, D), (C, D), (C, D)] * 2
+        actions += [(D, D), (D, D)]
+        # 20 D have passed (first isn't record)
+        actions += [(D, D)] * 100
+        # The defect streak will always be detected from here on, because it
+        # doesn't reset.  This logic comes before parity streaks or the turn-
+        # based logic.
+        self.versus_test(axelrod.Defector(), expected_actions=actions, attrs={"recorded_defects": 119})
+
+        # Detect random
+        expected_actions = [(C, D), (D, C), (C, D), (D, C), (C, D), (C, D), (D, D),
+                   (D, C), (C, D), (D, C), (C, C), (C, D), (D, D), (D, C),
+                   (C, D), (D, D), (D, C), (C, C), (C, D), (D, C), (C, D),
+                   (D, D), (D, C), (C, D), (D, D), (D, D), (C, D), (D, C),
+                   (C, C)]
+        # Enter defect mode.
+        expected_actions += [(D, C)]
+        random.seed(10)
+        player = self.player()
+        match = axelrod.Match((player, axelrod.Random()), turns=len(expected_actions))
+        # The history matrix will be [[0, 2], [5, 6], [3, 6], [4, 2]]
+        actions = match.play()
+        self.assertEqual(actions, expected_actions)  # Just to be consistant with the current test.
+        self.assertEqual(round(player.calculate_chi_squared(len(expected_actions)),3), 2.395)
```

Comment:
space after comma (before 3); drop the round or use `assertAlmostEqual` (which allows you to specify a number of places for equality):
```
self.assertAlmostEqual(a, b, places=4)
``` 

[Link to the docs](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertAlmostEqual) for `assertAlmostEqual`.